WasteWar/Assets/Scripts/Drawing/TerrainGridMesh.cs

```diff
@@ -0,0 +1,59 @@
+﻿using UnityEngine;
+using System.Collections.Generic;
+using Constants;
+
+//https://gist.github.com/mdomrach/a66602ee85ce45f8860c36b2ad31ea14
+
+
+public class TerrainGridMesh : MonoBehaviour
+{
+    public Terrain terrain;
+    //can make it visible in the inspector if necessary but it shouldn't be because the grid covers the whole terrain
+    private int GridSize;
+    private float cellLoc = 0f;
+
+    void Start()
+    {
+        GridSize= (int)(terrain.GetComponent<Terrain>().terrainData.size.x);
+
+        var mesh = new Mesh();
+        var verticies = new List<Vector3>();
+        var indicies = new List<int>();
+
+        GenerateLines(verticies,indicies);
+        DrawLines(mesh, verticies, indicies);
+    }
+
+    private void GenerateLines(List<Vector3> verticies, List<int> indicies)
+    {
+        //logic @ https://i.imgur.com/4n2iKqP.gifv
+        for (int i = 0; i <= (int)(GridSize / GridConstants.Instance.FloatCellSize()); i++)
+        {
+            verticies.Add(new Vector3(cellLoc, 0, 0));
+            verticies.Add(new Vector3(cellLoc, 0, GridSize));
+
+            indicies.Add(4 * i + 0);
```

Comment:
what is this 4

Reply:
TL:DR won't be reused anywhere but it is necessary for formation of lines between pairs of vertices, if you wanna make a const and name it, suggest me a name

it indicates the indexes of the vertices between which the mesh line will form,

so as an example,  i=0 
line 32 we add vertex 0
line 33 we add vertex 1

indicies.Add(4 * i + 0);
 indicies.Add(4 * i + 1);
creates line between vertex 0 and 1

then 41,42 we add vertex 2,3
indicies.Add(4 * i + 2);
indicies.Add(4 * i + 3);
creates line between vertex 2 and 3

without the 4, for i=1 the next set of lines would be incorrect